api/src/controllers/profile.ts

```diff
@@ -67,15 +73,20 @@ export const fetchAllProjectProfiles = async (req: Request, res: Response): Prom
 };
 
 export const fetchProjectProfile = async (
-  { params }: { params: any }, res: Response
+  { params, user }: { params: any, user: AuthenticatedUser }, res: Response
 ): Promise<void> => {
   const { ProfileModel } = dm;
   const { profileId } = params;
 
   try {
     const results = await ProfileModel.findById(Number(profileId));
 
-    res.status(200).json(results);
+    if (user.id === results.userId || user.roles.includes('administrator')) {
```

Comment:
I generally prefer early returns to if/else where possible; I think it makes the code more readable and less error prone. For example, one could:
```
if (user.id !== results.userId || !user.roles.includes('administrator')) {
   throw errorWithCode('Unauthorized Access', 401);
}
```

And skip the else.